### PR TITLE
Allow for unknown OIDs in extensions in wolfSSL_X509_set_ext()

### PR DESCRIPTION
**Pull request by anhu**
...and add some testing to show we properly inserted the extensions.

# Testing

Added tests to verify that extensions were inserted properly. 

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by anhu** [comment]
retest please

File: src/x509.c [hunk 1]
```diff
@@ -687,16 +687,9 @@ WOLFSSL_X509_EXTENSION* wolfSSL_X509_set_ext(WOLFSSL_X509* x509, int loc)
...
         ext->obj = wolfSSL_OBJ_nid2obj(nid);
-        if (ext->obj == NULL) {
-            WOLFSSL_MSG("\tfail: Invalid OBJECT");
-            wolfSSL_X509_EXTENSION_free(ext);
-            FreeDecodedCert(cert);
-        #ifdef WOLFSSL_SMALL_STACK
-            XFREE(cert, NULL, DYNAMIC_TYPE_DCERT);
-        #endif
-            return NULL;
+        if (ext->obj != NULL) {
```
Comment (JacobBarthelmeh) [inline src/x509.c:690]: Continuing on without a break if ext->obj is null results in potential null derefrences. An example would be if wolfSSL_OBJ_nid2obj failed and returned null and the case BASIC_CA_OID is used.

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]

File: src/x509.c (hunk 1 above)
Comment (anhu) [reply src/x509.c]: Woah!!! Sharp eye @JacobBarthelmeh !!  Thank you very much for spotting that. Its been fixed.

**Review by JacobBarthelmeh — APPROVED** [review]
Please follow up on this with adding WOLFSSL_CUSTOM_OID to automated CI tests if not already there.

**Pull request merged**